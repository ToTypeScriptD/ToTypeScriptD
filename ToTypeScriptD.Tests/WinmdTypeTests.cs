using ApprovalTests;
using Xunit;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Mono.Cecil;

namespace ToTypeScriptD.Tests
{

    public class WinmdTypeTests : WinmdTestBase
    {
        private TypeDefinition GetNativeType(string postfix)
        {
            return base.NativeAssembly.GetNativeType("ToTypeScriptD.Native." + postfix);
        }

        private TypeDefinition GetWinNativeType(string type)
        {
            return base.WindowsAssembly.GetNativeType(type);
        }

        [Fact]
        public void EnumType()
        {
            var result = GetNativeType("SampleEnum").ToTypeScript();
            Approvals.Verify(result);
        }

        [Fact]
        public void EnumTypeWithExplicitValues()
        {
            var result = GetNativeType("SampleEnumNumbered").ToTypeScript();
            Approvals.Verify(result);
        }

        [Fact]
        public void ClassWithEventHandler()
        {
            var result = GetNativeType("ClassWithEventHandler").ToTypeScript();
            Approvals.Verify(result);
        }

        [Fact]
        public void FullSampleAssembly()
        {
            var result = ToTypeScriptD.Render.FullAssembly(base.NativeAssembly.ComponentPath);
            Approvals.Verify(result);
        }

        [Fact]
        public void WindowsStorageClass()
        {
            var result = GetWinNativeType("Windows.Storage.StorageFile").ToTypeScript();
            Approvals.Verify(result);
        }
        [Fact]
        public void WindowsSystemUserProfileGlobalizationPreferencesClass()
        {
            var result = GetWinNativeType("Windows.System.UserProfile.GlobalizationPreferences").ToTypeScript();
            Approvals.Verify(result);
        }

        [Fact]
        public void IfEnumNamesAreAllCapsThenTheTranslateToAllLowerCase()
        {
            var result = GetWinNativeType("Windows.Foundation.Metadata.ThreadingModel")
                .ToTypeScript();
            Approvals.Verify(result);
        }

        [Fact]
        public void FullWindowsAssembly()
        {
            var file = @"C:\Windows\System32\WinMetadata\Windows.Foundation.winmd";
            var result = ToTypeScriptD.Render.FullAssembly(file);
            Approvals.Verify(result);
        }

        [Fact]
        public void AllWinmdFilesInWinMetadata()
        {

            var allFiles = System.IO.Directory.GetFiles(@"C:\Windows\System32\WinMetadata\", "*.winmd");

            var result = allFiles;
            Approvals.VerifyAll(result, "File: ");
            //.Verify();
        }
    }
}
