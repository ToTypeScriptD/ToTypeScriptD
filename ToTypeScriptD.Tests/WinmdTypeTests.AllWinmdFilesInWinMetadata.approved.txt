File: [0] = C:\Windows\System32\WinMetadata\Windows.ApplicationModel.winmd
File: [1] = C:\Windows\System32\WinMetadata\Windows.Data.winmd
File: [2] = C:\Windows\System32\WinMetadata\Windows.Devices.winmd
File: [3] = C:\Windows\System32\WinMetadata\Windows.Foundation.winmd
File: [4] = C:\Windows\System32\WinMetadata\Windows.Globalization.winmd
File: [5] = C:\Windows\System32\WinMetadata\Windows.Graphics.winmd
File: [6] = C:\Windows\System32\WinMetadata\Windows.Management.winmd
File: [7] = C:\Windows\System32\WinMetadata\Windows.Media.winmd
File: [8] = C:\Windows\System32\WinMetadata\Windows.Networking.winmd
File: [9] = C:\Windows\System32\WinMetadata\Windows.Security.winmd
File: [10] = C:\Windows\System32\WinMetadata\Windows.Storage.winmd
File: [11] = C:\Windows\System32\WinMetadata\Windows.System.winmd
File: [12] = C:\Windows\System32\WinMetadata\Windows.UI.winmd
File: [13] = C:\Windows\System32\WinMetadata\Windows.UI.Xaml.winmd
File: [14] = C:\Windows\System32\WinMetadata\Windows.Web.winmd
